lib/opentuna-stack.ts

```diff
@@ -147,5 +162,52 @@ export class OpentunaStack extends cdk.Stack {
       tunaManagerALBTargetGroup: tunaManagerStack.managerALBTargetGroup,
     });
     tunaManagerSG.connections.allowFrom(externalALBSG, ec2.Port.tcp(80), 'Allow external ALB to access tuna manager');
+
+    let commonBehaviorConfig = {
+      // special handling for HTTPS forwarding
+      forwardedValues: {
+        headers: ['Host', 'CloudFront-Forwarded-Proto'],
+        queryString: true,
+      },
+    };
+
+    // CloudFront as cdn
+    let cloudfrontProps = {
+      originConfigs: [{
+        customOriginSource: {
+          domainName: useHTTPS ? `${stack.region}.${domainName}` : externalALB.loadBalancerDnsName,
+        },
+        behaviors: [{
+          ...commonBehaviorConfig,
+          isDefaultBehavior: true,
+          // default 1 day cache
+          defaultTtl: cdk.Duration.days(1),
+        }, {
+          ...commonBehaviorConfig,
+          // 5min cache for tunasync status
+          pathPattern: '/jobs',
+          defaultTtl: cdk.Duration.minutes(5),
+        }],
+      }],
+    } as cloudfront.CloudFrontWebDistributionProps;
+    if (cloudfrontCert) {
+      // when https is enabled
+      cloudfrontProps = {
+        aliasConfiguration: {
```

Comment:
For China regions, the ACM does not support cloudfront yet. I will push new commits to use the SSL cert in IAM.